### PR TITLE
Extend lab-d with the provider dependency and e2e claim

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Extend lab-d with the provider dependency and e2e claim

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
k get claim
NAME                                                  SYNCED   READY   CONNECTION-SECRET             AGE
cluster.aws.platformref.upbound.io/platform-ref-aws   True     True    platform-ref-aws-kubeconfig   38m
```

### CODE REVIEW

File: advanced-compositions/lab-d-solution/providers.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-aws-iam
+spec:
+  package: xpkg.upbound.io/upbound/provider-aws-iam:v0.42.0
```
Comment (stevendborrelli) [inline advanced-compositions/lab-d-solution/providers.yaml:6]: Should this provider have a ControllerConfig?

File: advanced-compositions/lab-d/providers.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-aws-iam
+spec:
+  package: xpkg.upbound.io/upbound/provider-aws-iam:v0.42.0
+---
```
Comment (stevendborrelli) [inline advanced-compositions/lab-d/providers.yaml:7]: same question about controllerConfigRef

**Review by stevendborrelli — APPROVED** [review]
LGTM, just the question about controllerConfigRefs

File: advanced-compositions/lab-d-solution/providers.yaml (hunk 1 above)
Comment (haarchri) [inline advanced-compositions/lab-d-solution/providers.yaml:6]: no controllerconfig for provider-aws-iam ?

Comment (ytsarev) [reply advanced-compositions/lab-d-solution/providers.yaml:6]: thanks for the catch!

**Comment by ytsarev** [comment]
amended controlleconfigref for iam provider, thanks for the catch, merging

**Pull request merged**